### PR TITLE
chore: bump shellcheck v0.8.0

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

Seems we used v0.7.0 shellcheck and the current stable version is v0.8.0.

#### Summary of Changes

1. use shellcheck v0.8.0
2. fix some info/warn after upgrade.
    - add quote to variable 
    - rename a variable, `FORCE_DELETE`, to prevent self assignment
    - ignore check for `eval ... $@`
    - use `$*` to replace `$(eval echo "$@")`

### CODE REVIEW

File: net/scripts/colo-node-onfree.sh [hunk 1]
```diff
@@ -4,15 +4,15 @@
 SOLANA_LOCK_FILE="${SOLANA_LOCK_FILE:?}"
 SECONDARY_DISK_MOUNT_POINT="${SECONDARY_DISK_MOUNT_POINT:?}"
 SSH_AUTHORIZED_KEYS="${SSH_AUTHORIZED_KEYS:?}"
-FORCE_DELETE="${FORCE_DELETE}"
+NEED_TO_FORCE_DELETE="${FORCE_DELETE}"
```
Comment (t-nelson) [inline net/scripts/colo-node-onfree.sh:7]: is this intended to be an unset/empty check instead?

Comment (t-nelson) [reply net/scripts/colo-node-onfree.sh:7]:
```suggestion
FORCE_DELETE="${FORCE_DELETE:?}"
```

**Review by t-nelson — DISMISSED** [review]
r+ fixing the intended empty check in https://github.com/solana-labs/solana/pull/27463/commits/2e286f7c1abfca77d9b6d07bd06ccbad73d94b11

thanks for doing the fixup in individual commits!

File: net/scripts/colo-node-onfree.sh (hunk 1 above)
Comment (yihau) [reply net/scripts/colo-node-onfree.sh:7]: yeah. Originally I guessed we didn't use `${FORCE_DELETE?}"` cuz maybe it's unset for some reason.
I just searched our code again and found it only used at https://github.com/solana-labs/solana/blob/master/net/scripts/colo-utils.sh#L242-L249
Maybe it is harmless to use `${FORCE_DELETE?}`

Comment (yihau) [reply net/scripts/colo-node-onfree.sh]: fixed [3935ed1](https://github.com/solana-labs/solana/pull/27463/commits/3935ed105058b793390a514173782021de561ee9)

**Pull request merged**